src/python/pants/backend/python/tasks/build_local_python_distributions.py

```diff
@@ -44,89 +45,104 @@ def product_types(cls):
+    return reqs_to_resolve
 
-    overrides = {
-      'purelib': site_dir,
-      'headers': os.path.join(site_dir, 'headers'),
-      'scripts': os.path.join(site_dir, 'bin'),
-      'platlib': site_dir,
-      'data': site_dir
-    }
+  # NB: these are all the immediate subdirectories of the target's results directory.
+  # This contains any modules from a setup_requires().
+  setup_requires_site_subdir = 'setup_requires_site'
+  # This will contain the sources used to build the python_dist().
+  dist_subdir = 'python_dist_subdir'
 
-    # The `python_dist` target builds for the current platform only.
-    for obj in setup_requires_dists['current']:
-      wf = WheelFile(obj.location)
-      wf.install(overrides=overrides, force=True)
+  @classmethod
+  def _get_output_dir(cls, results_dir):
+    return os.path.join(results_dir, cls.dist_subdir)
 
-    return site_dir
+  @classmethod
+  def _get_dist_dir(cls, results_dir):
+    return os.path.join(cls._get_output_dir(results_dir), SetupPyRunner.DIST_DIR)
 
   def execute(self):
-    dist_targets = self.context.targets(self.filter_target)
-
-    if dist_targets:
-      with self.invalidated(dist_targets,
-                            fingerprint_strategy=DefaultFingerprintStrategy(),
-                            invalidate_dependents=True) as invalidation_check:
-        interpreter = self.context.products.get_data(PythonInterpreter)
-
-        for vt in invalidation_check.invalid_vts:
-          if vt.target.dependencies:
-            raise TargetDefinitionException(
-              vt.target, 'The `dependencies` field is disallowed on `python_dist` targets. '
-                         'List any 3rd party requirements in the install_requirements argument '
-                         'of your setup function.'
-            )
-          setup_req_dir = os.path.join(vt.results_dir, 'setup_requires_site')
-          pythonpath = self._ensure_setup_requires_site_dir(dist_targets, interpreter, setup_req_dir)
-          self._create_dist(vt.target, vt.results_dir, interpreter, pythonpath)
-
-        local_wheel_products = self.context.products.get('local_wheels')
-        for vt in invalidation_check.all_vts:
-          dist = self._get_whl_from_dir(os.path.join(vt.results_dir, 'dist'))
-          req_lib_addr = Address.parse('{}__req_lib'.format(vt.target.address.spec))
-          self._inject_synthetic_dist_requirements(dist, req_lib_addr)
-          # Make any target that depends on the dist depend on the synthetic req_lib,
-          # for downstream consumption.
-          for dependent in self.context.build_graph.dependents_of(vt.target.address):
-            self.context.build_graph.inject_dependency(dependent, req_lib_addr)
-          local_wheel_products.add(vt.target, os.path.dirname(dist)).append(os.path.basename(dist))
+    dist_targets = self.context.targets(is_local_python_dist)
+    if not dist_targets:
```

Comment:
A reminder that this type of edit (dedenting a bunch of stuff) makes the diff useless... best to make this kind of change in reviews where relatively little else is going on.

Reply:
Reverted! Will remember for the future.